File: src/js/core/src/Error/CircularColorReferenceError.ts
/**
 * Thrown when a color in the style definition references itself, directly or
 * indirectly. The {@link chain} field reproduces the resolution path.
 */
export class CircularColorReferenceError extends Error {
  readonly chain: readonly string[];

  constructor(chain: readonly string[]) {
    const path = chain.join(' → ');

    super(`Circular color reference: ${path}`);
    this.name = 'CircularColorReferenceError';
    this.chain = chain;
  }
}

File: src/dart/core/lib/src/error/circular_color_reference_error.dart
/// Thrown when a color in the style definition references itself, directly
/// or indirectly. [chain] reproduces the resolution path.
class CircularColorReferenceError implements Exception {
  /// The resolution path that closed the cycle, e.g. `[a, b, a]`.
  final List<String> chain;

  /// The formatted error message.
  final String message;

  CircularColorReferenceError(List<String> chain)
      : chain = List.unmodifiable(chain),
        message = 'Circular color reference: ${chain.join(' → ')}';

  @override
  String toString() => message;
}
